src/protocols/bdx/BdxTransferSession.h

```diff
@@ -0,0 +1,332 @@
+     *   Prepare a BlockQuery message. The Block counter will be populated automatically.
+     *
+     * @return CHIP_ERROR The result of the preparation of a BlockQuery message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR PrepareBlockQuery();
+
+    /**
+     * @brief
+     *   Prepare a Block message. The Block counter will be populated automatically.
+     *
+     * @param inData Contains data for filling out the Block message
+     *
+     * @return CHIP_ERROR The result of the preparation of a Block message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR PrepareBlock(const BlockData & inData);
+
+    /**
+     * @brief
+     *   Prepare a BlockAck message. The Block counter will be populated automatically.
+     *
+     * @return CHIP_ERROR The result of the preparation of a BlockAck message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR PrepareBlockAck();
+
+    /**
+     * @brief
+     *   Prematurely end a transfer. Must still call Reset() to prepare the TransferSession for another transfer.
+     *
+     * @param reason The StatusCode reason for ending the transfer.
+     *
+     * @return CHIP_ERROR May return an error if there is no transfer in progress.
+     */
+    CHIP_ERROR AbortTransfer(StatusCode reason);
+
+    /**
+     * @brief
+     *   Reset all TransferSession parameters. The TransferSession object must then be re-initialized with StartTransfer() or
+     *   WaitForTransfer().
+     */
+    void Reset();
+
+    /**
+     * @brief
+     *   Process a message intended for this TransferSession object.
+     *
+     * @param msg       A PacketBufferHandle pointing to the message buffer to process. May be BDX or StatusReport protocol.
+     * @param curTimeMs Current time indicated by the number of milliseconds since some epoch defined by the platform
+     *
+     * @return CHIP_ERROR Indicates any problems in decoding the message, or if the message is not of the BDX or StatusReport
+     *                    protocols.
+     */
+    CHIP_ERROR HandleMessageReceived(System::PacketBufferHandle msg, uint64_t curTimeMs);
+
+    TransferControlFlags GetControlMode() const { return mControlMode; }
+    uint64_t GetStartOffset() const { return mStartOffset; }
+    uint64_t GetTransferLength() const { return mTransferLength; }
+    uint16_t GetTransferBlockSize() const { return mTransferMaxBlockSize; }
+
+    TransferSession();
+
+private:
+    enum TransferStates : uint8_t
+    {
+        kState_Idle,
+        kState_AwaitingInitMsg,
+        kState_AwaitingAccept,
+        kState_NegotiateTransferParams,
+        kState_TransferInProgress,
+        kState_AwaitingEOFAck,
+        kState_ReceivedEOF,
+        kState_TransferDone,
+        kState_Error,
+    };
```

Comment:
Suggest removing "kState_" prefix and renaming enum to `TransferState` now that it's non-overlapping

Reply:
done